llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -2806,7 +2806,8 @@ static bool hoistBOAssociation(Instruction &I, Loop &L,
     return false;
 
   auto *BO0 = dyn_cast<BinaryOperator>(BO->getOperand(0));
-  if (!BO0 || BO0->getOpcode() != Opcode || !BO0->isAssociative())
+  if (!BO0 || BO0->getOpcode() != Opcode || !BO0->isAssociative() ||
+      BO0->getNumUses() > 2)
```

Comment:
```suggestion
      BO0->hasNUsesOrMore(3))
```
Early exit use list scan.

Reply:
Thanks, done.